test/ring.lean

```diff
@@ -125,3 +125,12 @@ example (a : Nat) : 1 * f a * 1 = f (a + 0) := by
 
 -- Powers in the exponent get evaluated correctly
 example (X : ℤ) : (X^5 + 1) * (X^2^3 + X) = X^13 + X^8 + X^6 + X := by ring
+
+-- simulate the type of MvPolynomial
+def R : Type u → Type v → Sort (max (u+1) (v+1)) := sorry
```

Comment:
But `MvPolynomial` in the docs is apparently of type `Type (maxu_1u_2)`...

Reply:
Right the delaborator is smarter than lean itself in a way, it also uses `Level.dec` to normalize the type.
If you print the decl with pp.raw you'll see
```
def MvPolynomial.{u_1, u_2} : Type.{u_1} -> (forall (R : Type.{u_2}) [inst._@.Mathlib.Data.MvPolynomial.Basic._hyg.24 : CommSemiring.{u_2} R], Sort.{max (succ u_1) (succ u_2)}) :=
fun (σ : Type.{u_1}) (R : Type.{u_2}) [inst._@.Mathlib.Data.MvPolynomial.Basic._hyg.24 : CommSemiring.{u_2} R] => AddMonoidAlgebra.{u_2, u_1} R (Finsupp.{u_1, 0} σ Nat (LinearOrderedCommMonoidWithZero.toZero.{0} Nat Nat.linearOrderedCommMonoidWithZero)) (CommSemiring.toSemiring.{u_2} R inst._@.Mathlib.Data.MvPolynomial.Basic._hyg.24)
```

Reply:
Can we just add an explicit type to AddMonoidAlgebra to fix that?

Reply:
I think if something is accepted by Lean tactics should be able to handle it, unless I misunderstand your suggestion.
Otherwise we are in a situation where users can easily footgun themselves by defining types that they cant run tactics on.

Reply:
Note that the level normalization automatically turns `(max u v) + 1` into `max (u+1) (v+1)`, so no amount of rewriting will change this. Using `.dec` is the correct solution.

Reply:
Oh interesting, I had the wrong mental model of what normalization should do here.

Reply:
As did I, this is not the direction I would have expected, it seems less useful.